components/brave_wallet/resources/solana_provider.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+(function () {
+  if (!window.solana) {
```

Comment:
i just wanted to confirm that this entire script will NOT be injected unless the feature flag (off by default) is enabled - https://github.com/brave/brave-core/pull/12554/files#diff-bd0508db5780666e635128717e778f2a51486f055feb9f236d03aefdb1dc122cR59-R60

because a page can define window.solana to true

Reply:
sorry i just realized you answered this here - https://github.com/brave/brave-core/pull/12554#discussion_r831354629

however is this the only guard we have against the rest of the script running if the feature flag is enabled? so if a page defines `window.solana` in a private window, this would run?

Reply:
it won't run in private window for both `window.ethereum` and `window.solana` because it early return here 
https://github.com/brave/brave-core/pull/12554/files#diff-bd0508db5780666e635128717e778f2a51486f055feb9f236d03aefdb1dc122cR43-R47
`brave_use_native_wallet` is decided by `brave_wallet::IsAllowedForContext` and user default wallet setting

Reply:
thx